src/axom/core/Array.hpp

```diff
@@ -956,18 +941,15 @@ inline typename Array<T, DIM, SPACE>::ArrayIterator Array<T, DIM, SPACE>::erase(
   Array<T, DIM, SPACE>::ArrayIterator pos)
 {
   assert(pos >= begin() && pos < end());
-  int counter = 0;
 
-  while(pos < end() - 1)
-  {
-    *pos = *(pos + 1);
-    pos += 1;
-    counter += 1;
-  }
-  (*pos).~T();
+  IndexType posIdx = pos - begin();
```

Comment:
For methods like this (insert above), I think we should clearly state in the methods documentation what element the returned iterator refers to.